common/djangoapps/course_modes/models.py

```diff
@@ -375,6 +403,128 @@ def min_course_price_for_currency(cls, course_id, currency):
+                enrollment_title = __("Your verification is pending")
+                enrollment_value = __("Verified: Pending Verification")
+                show_image = True
+                image_alt = __("ID verification pending")
+            elif verification_status == VERIFY_STATUS_APPROVED:
+                enrollment_title = __("You're enrolled as a verified student")
+                enrollment_value = __("Verified")
+                show_image = True
+                image_alt = __("ID Verified Ribbon/Badge")
+            else:
+                enrollment_title = __("You're enrolled as an honor code student")
+                enrollment_value = __("Honor Code")
+        elif mode == "honor":
+            enrollment_title = __("You're enrolled as an honor code student")
+            enrollment_value = __("Honor Code")
+        elif mode == "audit":
+            enrollment_title = __("You're auditing this course")
+            enrollment_value = __("Auditing")
+        elif mode in["professional", "no-id-professional"]:
+            enrollment_title = __("You're enrolled as a professional education student")
+            enrollment_value = __("Professional Ed")
+
+        return {
+            'enrollment_title': enrollment_title,
+            'enrollment_value': enrollment_value,
+            'show_image': show_image,
+            'image_alt': image_alt,
+            'display_mode': CourseMode._get_certificate_display_mode(mode, verification_status)
+        }
+
+    @staticmethod
+    def _get_certificate_display_mode(enrollment_mode, verification_status):
+        """
+         return the of mode after checking enrollment mode and status.
+         Args:
+            mode (str): enrollment mode.
+            verification_status (str) : verification status of student
+
+        Returns:
+            str: display mode for certs
+        """
+
+        # import inside the function to avoid the circular import
+        from student.helpers import (
+            VERIFY_STATUS_NEED_TO_VERIFY,
+            VERIFY_STATUS_SUBMITTED,
+            VERIFY_STATUS_APPROVED
+        )
+
+        if enrollment_mode == "verified":
+            if verification_status in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
+                mode = "verified"
+            else:
+                mode = "honor"
+        elif enrollment_mode == "professional" or enrollment_mode == "no-id-professional":
+            mode = "professional"
+        else:
+            mode = enrollment_mode
+
+        return mode
+
+    @classmethod
+    def has_professional_mode(cls, modes_dict):
+        """
+        check the course mode is profession or no-id-professional
+
+        Args:
+            modes_dict (dict): course modes.
+
+        Returns:
+            bool
+        """
+        return "professional" in modes_dict or CourseMode.NO_ID_PROFESSIONAL_MODES[0] in modes_dict
```

Comment:
Since this is a class method, you should access `NO_ID_PROFESSIONAL_MODES` using `cls.NO_ID_PROFESSIONAL_MODES[0]`. Indexing to the first item in the list also makes me uncomfortable, since it assumes that the first item in that list will always be `'no-id-professional'`. It seems like you might benefit from either use of a constant containing the professional modes, or at least use of constants for each mode string, and checking their membership in the `modes_dict` explicitly.

In addition, this is a nitpick, but I've noticed frequent switching between using single quotes and double quotes for strings. Our general rule is to use double quotes for copy strings, and single quotes for everything else.